### DISCUSSION TITLE
How to run Next Js application build (out) directory in local?

### DISCUSSION THREAD

**Discussion by Maniraj-Murugan** [vercel/next.js#16439] [Help] [answered]
I have done development of Next Js application and as of now I have done auto deployment using https://vercel.com/

Things are fine as of now.. But here came the requirement that I need to build the Next Js application and share the build folder with the team for deployment in server.

The commands I have followed,

```npm run build``` &
```npm run export```

And the above one creates the ```out``` directory.. So how to run this out directory in my local machine to check whether the build folder is working as expected before sharing with the team?

Folder structure of ```out``` directory:

     -> out
    
          -> _next
                 -> static
                 -> xxxxxxxxxxxxx (some random name)
          -> static
          -> home.png
          -> location.png

So anyone could kindly please help me how can I run this generated build folder (out) to check whether the developed Next Js application works fine in my local machine after which I can share the same built folder to the team?

While using vercel things happen automatically whenever we push the code, it  deploy automatically, but I am not sure how actually deployment happen there.. In other frameworks we used to get the dist folder with index.html which makes easy to understand for deployment.

I am in the need to confirm that build folder works fine in my local before sharing it with team for deployment. So for that I need to build and run the Next Js app in my local..

**Comment by jamesmosier** [comment, marked as answer]
You could use [`serve`](https://github.com/vercel/serve#usage) locally to test it out.

```bash
serve out
```

**Reply by Maniraj-Murugan** [reply]
It throws the following error,

``` 
serve : The term 'serve' is not recognized as the name of a  cmdlet, function, script file, or operable program. Check the     
spelling of the name, or if a path was included, verify that the  
path is correct and try again. 
```

**Reply by jamesmosier** [reply]
Did you install it? `npm install -g serve`? And then you might have to open up a new terminal session in order for the command to be recognized.

**Reply by Maniraj-Murugan** [reply]
I have tried like above step which you have said but it only gives the files that are available in root level of the application and the UI of the app is not rendered.

**Reply by jamesmosier** [reply]
Hmm I'm not sure then. I have an exportable Next app that I run `next build && next export` and that produces an `out` folder and then I run `serve out` and to go `localhost:5000` and I see my app 🤔

**Reply by thisisgopal** [reply]
```
$ npm i -g serve
$ serve out
```
It works !

**Reply by gopalwebdev** [reply]
npm i -g serve
serve -s out // if you wanna run react production build try the below one.
serve -s build

**Reply by NickKarsant** [reply]
what is the "-s" for and why is it not explained in the docs for serve?

**Reply by well1791** [reply]
is `npm run export` still valid given the latest versions of nextjs? 

EDIT: looks like it's just `npm run build && npx next start -p 4000`  https://nextjs.org/docs/app/api-reference/next-cli#production

**Reply by lucas-0liveira** [reply]
nice

**Reply by hedleysmith** [reply]
`npx serve out`

**Reply by junaidhnhtechsolutions** [reply]
npm i -g serve
serve -s out // if you wanna run react production build try the below one.
serve -s build

**Reply by den-ii** [reply]
I can't see my images when I do this
